backend/curation/api/curation-api.yml

```diff
@@ -14,6 +14,27 @@ info:
     ## Data Model
     <!-- Data model diagram created via Lucid: https://lucid.app/lucidchart/8d085832-34f4-4413-9c1d-44e77d06e326/edit?invitationId=inv_ed354532-7256-4e0a-9ad0-b622905e5198&page=Vfzkzdv~I1fL# -->
     <img src="/static/discover-api-data-model.png" width="50%"/>
+
+    ## Working with versions
+
+    #### Access to prior published version history
+    The CZ CELLxGENE Discover API employs the concept of *Versions* for our two core entities, *Collections* and
+    *Datasets*. This is to permit access to prior published instances (versions) of these entities even as new updates
+    for them are published.
+
+    #### Version identifiers
+    While the *Collection* identifier `collection_id` will always refer to the latest published instance of a Collection
+    (like the "head" in version-control terminology), and must be used with `Collection` endpoints, the
+    *Collection Version* identifier `collection_version_id` points always to one specific published instance (version)
+    of a Collection (like a commit hash) and must be used with `Collection Version` endpoints.
+    endpoints. The same is true for *Datasets* and *Dataset Versions*, though it is important to note that if a given Dataset
+    is not updated during the course of a revision of its parent Collection, then its *Dataset Version* identifier
```

Comment:
nit: I like this better, but I still think it's a bit clunky talking about the identifier changing or not changing instead of saying a Dataset Version is persisted or a new one is created. It's not just the identifier that changes if a Dataset has a new Version, so it feels incomplete to hone in on that and it's implied that the version ID will stay the same if it's the same Dataset Version.

Reply:
I've tried to split the difference and refer to both 🙂